plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/client/AwsRequestSigner.java

```diff
@@ -39,22 +39,24 @@
 
 import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.plugin.opensearch.AwsSecurityConfig.DeploymentType;
 import static java.lang.String.CASE_INSENSITIVE_ORDER;
 import static org.apache.http.protocol.HttpCoreContext.HTTP_TARGET_HOST;
 
 class AwsRequestSigner
         implements HttpRequestInterceptor
 {
-    private static final String SERVICE_NAME = "aoss";
+    private final String serviceName;
     private final AWSCredentialsProvider credentialsProvider;
     private final AWS4Signer signer;
 
-    public AwsRequestSigner(String region, AWSCredentialsProvider credentialsProvider)
+    public AwsRequestSigner(String region, DeploymentType deploymentType, AWSCredentialsProvider credentialsProvider)
     {
         this.credentialsProvider = credentialsProvider;
         this.signer = new AWS4Signer();
+        this.serviceName = deploymentType == DeploymentType.OPENSEARCH_SERVERLESS ? "aoss" : "es";
```

Comment:
Use a switch statement so the code is future-proof:

```java
this.serviceName = switch (deploymentType) {
    case SERVERLESS -> ...;
    case PROVISIONED -> ...;
};
```

Reply:
Agreed.